jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/CharsetStringBuilder.java

```diff
@@ -128,6 +176,42 @@ public void append(byte b)
             _buffer.put(b);
         }
 
+        @Override
+        public void append(char c)
+        {
+            try
+            {
+                if (_stringBuilder == null)
+                    _stringBuilder = new StringBuilder();
+                _stringBuilder.append(_decoder.decode(_buffer.flip()));
+                _buffer.clear();
+            }
+            catch (CharacterCodingException e)
+            {
+                // Should do replacement and remember issue.
```

Comment:
This is the responsibility of the `CharsetDecoder` and how it's configured, we should honor its `onalformedInput` configuration.

Reply:
Yep, didn't have time to work it out exactly. But yes, the decoder should do the replacement, not us.

Reply:
The behaviour was correct, so I have just updated the comment.